src/System.Linq/src/System/Linq/Concat.cs

```diff
@@ -498,6 +498,29 @@ public List<TSource> ToList()
                 return list;
             }
 
+            public HashSet<TSource> ToHashSet(IEqualityComparer<TSource> comparer)
+            {
+                int count = GetCount(onlyIfCheap: true);
+                var hashSet = count != -1 ? new HashSet<TSource>(count, comparer) : new HashSet<TSource>(comparer);
```

Comment:
This will waste a lot of memory if there are many duplicates. (I won't keep repeating this).